server/controllers/conversation.js

```diff
@@ -0,0 +1,106 @@
+const mongoose = require("mongoose");
+const Conversation = require("../models/Conversation");
+const asyncHandler = require("express-async-handler");
+
+// @route GET /conversations
+// @desc Get all conversations of a logged-in user
+// @access private
+exports.getAllConversations = asyncHandler(async (req, res, next) => {
+  try {
+    const userId = req.user.id;
+
+    const conversations = await Conversation.find({
+      $or: [{ user1: userId }, { user2: userId }],
+    })
+      .populate("messages")
```

Comment:
Wouldn't it be better to only get the last message and read status of that message when we fetch all conversations here (since we are already populating the messages for a single conversation within the `getConversation` controller) 😄

Reply:
Applied 🙂